AmplifyPlugins/Core/AWSPluginsCore/Model/Support/SelectionSet.swift

```diff
@@ -8,104 +8,102 @@
 import Foundation
 import Amplify
 
-/// Contains detail about the current selection set. `pagination` indicates that it contains paginated fields.
-/// `modelField` indicates that the values represent the property names from a `ModelSchema`.
-public enum SelectionSetType {
-    case pagination
-    case modelField
-}
+public typealias SelectionSet = TreeNode<SelectionSetField>
 
-/// A nested structure to contain a list of `SelectionSetField`, each of which may optionally contain another
-/// `SelectionSet`
-public struct SelectionSet {
-    var fields: [SelectionSetField]
-    var type: SelectionSetType
+public class TreeNode<E> {
```

Comment:
1. Put `TreeNode` in its own file
1. `TreeNode` is a public class, but its methods and properties are internal. Intentional?
1.  Since we're naming a type, we should probably call it [`Tree` instead of `TreeNode`](https://en.wikipedia.org/wiki/Tree_(data_structure)#Data_type_versus_data_structure)

Reply:
updating 1 and 3, for 2, `Tree` is pubic for the `typealias SelectionSet` as SelectionSet is used as a property on the `SingleDirectiveGraphQLDocument` public protocol. Everything else on the Tree is used internally.